### ISSUE TITLE
Collisions with NPCs and Objects

### ISSUE THREAD

**Issue by robert-beall** [jdoiron94/Outside-The-Box#6]
Currently, levitated objects pass through NPCs and Objects. 

Levitated Objects must collide with other objects/entities.

**Comment by jdoiron94** [comment]
Liftable objects currently only clip with the player and NPCs. It would only take a few lines to fix that.

**Comment by jdoiron94** [comment]
Also, the player/npc collisions will need to be fixed to prevent the player from ramming into the npc and stopping its pathing and from locking the player out of moving in certain directions.

**Comment by jdoiron94** [comment]
Fixed liftable objects colliding with all entities. I left it to allow colliding with tokens. If we changed that, it'd be cumbersome for the player to move objects around and most games allow collisions with tokens. We still need to figure out why it's possible to stop the ai path by pushing against them and why throwable objects/the player get stuck and aren't allowed to move in valid directions when in the ai path. Additionally, the dash ability needs to be checking for collisions before it moves the player.
